cpp/src/io/csv/csv_gpu.cu

```diff
@@ -999,14 +995,8 @@ __global__ void __launch_bounds__(rowofs_block_dim)
       rowmap >>= pos;
     }
     // Return the number of rows out of range
-    rows_out_of_range = half_warp_reduce(temp_storage.half[t / 32]).Sum(rows_out_of_range);
-    __syncthreads();
-    if (!(t & 0xf)) { ctxtree[t >> 4] = rows_out_of_range; }
-    __syncthreads();
-    if (t < 32) {
-      rows_out_of_range = warp_reduce(temp_storage.full).Sum(static_cast<uint32_t>(ctxtree[t]));
-      if (t == 0) { row_ctx[blockIdx.x] = rows_out_of_range; }
-    }
+    rows_out_of_range = block_reduce(temp_storage).Sum(rows_out_of_range);
+    if (t == 0) { row_ctx[blockIdx.x] = rows_out_of_range; }
```

Comment:
I've stared at this for a while now and cannot think of any reason why this would be slower except for potentially larger shared memory usage.
In the old code `ctxtree `was reused to store per-warp reduction results, which (I think) means that `ctxtree` and `temp_storage` are not used at the same time and we might be able to put them in a union.